packages/core/src/RPA/core/windows/window.py

```diff
@@ -87,10 +89,14 @@ def list_windows(
             rectangle = (
                 [rect.left, rect.top, rect.right, rect.bottom] if rect else [None] * 4
             )
+            if self.ctx.list_processes:
+                name = process_list[pid] if pid in process_list else None
+            else:
+                name = Path(fullpath).name if fullpath else None
```

Comment:
```suggestion
            name = process_list[pid] if pid in process_list else None
            if not name and fullpath:
                name = Path(fullpath).name
```
Just in case you feel is better to try to retrieve a `name` from everywhere we can, no matter the circumstances and expectations.

Reply:
Yeah maybe but as this is **this** or that **setting** then your suggestion would do 2 operations in the case we really just want to do one.

Reply:
Ah good point. I will change this then.